src/condor_utils/file_transfer.cpp

```diff
@@ -4141,7 +4144,37 @@ FileTransfer::DoUpload(filesize_t *total_bytes_ptr, ReliSock *s)
 		}
 	}
 
+
+	//
+	// Remove file entries which result in duplicates at the destination.
+	// This is no longer necessary for correctness (because I changed
+	// FileTransferItem::operator < to group rather than sort), but it's
+	// still more efficient.  Some of these duplicates will have been
+	// generated internally, but not all.  It's more efficient to remove
+	// them all here (especially when the duplicate would only have been
+	// generated by expanding a directory).  We remove duplicate directory
+	// creation entries when we create the extra entries because for those,
+	// we want to preserve the earlier one, rather than the later one.
+	//
+	std::set<std::string> names;
+	for( auto iter = filelist.rbegin(); iter != filelist.rend(); ++iter ) {
+		auto & item = * iter;
+
+		if( item.isSrcUrl() ) { continue; }
+		if( item.isDestUrl() ) { continue; }
+
+		std::string rd_path = item.destDir() + DIR_DELIM_CHAR + condor_basename(item.srcName().c_str());
+		if( names.find(rd_path) == names.end() ) {
+			names.insert(rd_path);
+		} else {
+			// This incancation converts a reverse to a forward iterator.
+			filelist.erase( (iter + 1).base() );
+		}
+	}
+	for( auto & i: filelist ) { dprintf( D_KEYBOARD, ">>> DoUpload(), file-item after duplicate removal: %s -> %s\n", i.srcName().c_str(), i.destDir().c_str() ); }
```

Comment:
D_KEYBOARD seems a really weird place to stash this.  Seems to be developer-only debugging; suggest remove.

Reply:
It's in D_KEYBOARD precisely because they won't be turned on by default.  I've commented them out rather than delete because I have no doubt they will be needed again.